contracts/CollectiveGovernance.sol

```diff
@@ -100,27 +111,27 @@ contract CollectiveGovernance is Governance, VoteStrategy, ERC165 {
     }
 
     modifier requireVoteReady(uint256 _proposalId) {
-        require(_storage.isFinal(_proposalId), "Voting is not ready");
+        if(!(_storage.isFinal(_proposalId)) revert CG__VotingNotReady();
```

Comment:
I'm not sure if you used a refactoring tool but please distribute the !'s through the logic and simplify all the cases.  Including the use of redundant parenthesis.